crates/rbuilder/src/building/builders/merging_builder/combinator.rs

```diff
@@ -30,12 +30,17 @@ pub struct CombinatorContext<DB> {
     coinbase_payment: bool,
     root_hash_config: RootHashConfig,
     builder_name: String,
+    phantom: PhantomData<DB>,
 }
 
-impl<DB: Database + Clone + 'static> CombinatorContext<DB> {
+impl<P, DB> CombinatorContext<P, DB>
+where
+    DB: Database + Clone + 'static,
+    P: DatabaseProviderFactory<DB> + StateProviderFactory + Clone + 'static,
```

Comment:
How does this work with Rbuilder as a module? AFAIK It only receives the StateProviderFactory, not DatabaseProviderFactory<DB>.

Reply:
It is also `DatabaseProviderFactory` https://github.com/paradigmxyz/reth/blob/661b260f6172c047e8530e2331b2c84141e03c2b/crates/storage/provider/src/traits/full.rs#L14